lib/chef/provider/powershell_script.rb

```diff
@@ -27,8 +27,71 @@ class PowershellScript < Chef::Provider::WindowsScript
       provides :powershell_script
 
       action :run do
-        validate_script_syntax!
-        super()
+        Chef::Log.debug("using inline impl: #{new_resource.use_inline_powershell}")
+        if new_resource.use_inline_powershell
```

Comment:
Now that I can see the forest, I will more thoughts. Stay tuned.

Reply:
Ok, so I think the general flow and logic is good, but I really would like to see this chunked up and modularized better.

Reply:
> Ok, so I think the general flow and logic is good, but I really would like to see this chunked up and modularized better.

I can definitely look at doing that - there are a couple of places I can see to do that:
a) moving the inline powershell version into a function, rather than in action :run, but 
b) modularizing the stuff i cut/pasted from execute - I avoided that because I didn't think it was a good idea to mess with execute.rb generally, BUT honestly, looking at it with fresh eyes, that seems like a sane thing to do?

Reply:
@jaymzjulian ok, the cut/paste modularization might be better served for a *second* PR, maybe even after the next Chef release. @dafyddcrosby What do you think about introducing this for the next release and then refactoring as a second pass?